### PR TITLE
Make user email case insensitive

### PR DESCRIPTION
**Pull request by avalmas-programize**
https://github.com/aqualinkorg/aqualink-app/issues/510

### CODE REVIEW

File: packages/api/src/users/users.service.ts [hunk 1]
```diff
@@ -27,7 +27,9 @@ export class UsersService {
     if (!firebaseUser) {
       throw new BadRequestException('Invalid Firebase token.');
     }
-    if (firebaseUser.email !== createUserDto.email) {
+    if (
```
Comment (ericboucher) [inline packages/api/src/users/users.service.ts:30]: So do we know what was happening? Firebase always return the email in lowercase?

**Comment by ericboucher** [comment]
@tleraj maybe we could also solve this in the frontend actually and make sure the email input are of type `email` to control the case.

File: packages/api/src/users/users.service.ts (hunk 1 above)
Comment (avalmas-programize) [reply packages/api/src/users/users.service.ts:30]: Yes all user emails are stored in firebase in lowercase

**Comment by avalmas-programize** [comment]
Since we plan to expose the backend API to the public we need have checks in the backend as well

**Comment by avalmas-programize** [comment]
Class/validator has a guard `IsLowercase` that we can use. Quite unexpected but also quite handy
So this will simplify the backend checks

File: packages/website/src/common/RegisterDialog/index.tsx [hunk 2]
```diff
@@ -64,7 +65,7 @@ const RegisterDialog = ({
       const registerInfo: UserRegisterParams = {
         fullName: `${data.firstName} ${data.lastName}`,
         organization: data.organization,
-        email: data.emailAddress,
+        email: data.emailAddress.toLowerCase(),
```
Comment (ericboucher) [inline packages/website/src/common/RegisterDialog/index.tsx:68]: Ideally we can only type lowercase (eg. ovio.org)

Or at least also transform to lowercase in the sign in

**Comment by tleraj** [comment]
@ericboucher With the latest change, the input value on the email text field is always transformed into lowercase (in both sign in and sign up dialogs) by using a `react-hook-form` workaround
```javascript
onChange={() =>
  setValue(
    "emailAddress",
    getValues("emailAddress").toLowerCase()
  )
}
```

Also the values in the request payloads on the submit functions are transformed into lowercase just in case.

**Review by tleraj — APPROVED** [review]
Looks good :+1:

**Review by ericboucher — CHANGES_REQUESTED** [review]
Let's make the backend safe instead of using guards

**Review by ericboucher — APPROVED** [review]

**Review by ericboucher — APPROVED** [review]

**Pull request merged**